tensorflow/core/kernels/data/name_utils.h

```diff
@@ -33,10 +33,13 @@ string OpName(const string& dataset_type);
 // Returns a human-readable debug string for this dataset in the format of
 // "FooDatasetOp(arg1, arg2, ...)::Dataset".
 //
-// e.g. DatasetDebugString("Map", {}) -> "MapDatasetOp::Dataset";
-// DatasetDebugString("Range", {"0", "10", "3"}) ->
-// "RangeDatasetOp(0, 10, 3)::Dataset".
+// e.g. DatasetDebugString("Map", "Dataset", {}) -> "MapDatasetOp::Dataset";
+// DatasetDebugString("Range", "Dataset", {"0", "10", "3"}) ->
+// "RangeDatasetOp(0, 10, 3)::Dataset";
+// DatasetDebugString("Shuffle", "FixedSeedDataset", {"10", "1", "2"}) ->
+// "ShuffleDatasetOp(10, 1, 2)::FixedSeedDataset";
 string DatasetDebugString(const string& dataset_type,
+                          const string& dataset_name,
```

Comment:
As `ShuffleDatasetOp` class has two private dataset classes `ReshufflingDataset` and `FixedSeedDataset`. Their debug strings are different with different dataset_name: "ShuffleDatasetOp(10, 1, 2)::ReshufflingDataset" vs. "ShuffleDatasetOp(10, 1, 2)::FixedSeedDataset". For this case, we need to make DatasetDebugString() function be more flexible to support different dataset name. 

Additionally, since most of the dataset Ops have only one dataset class and their names are `Dataset`, I make the change here instead of the other DatasetDebugString function: `string DatasetDebugString(const string& dataset_type, const Args&... args)` in order to avoid breaking other usages.

Reply:
As per my earlier comment, I would prefer the overload to introduce a prefix argument (instead of a name argument).